### ISSUE TITLE
Reproducable panic in MacOS usage

### ISSUE THREAD

**Issue by tharwerth** [RussHewgill/purgedb#1]
From the "Enter Purge Values" screen, if there is only a value in one of the "Swap" fields (IE only one filament exists in the db), selecting black will always lead to a crash

```
thread 'main' panicked at src/gui/enter_purge.rs:34:74:
called `Result::unwrap()` on an `Err` value: QueryReturnedNoRows
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/macos/app_state.rs:387:33:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }

```

Once a second filament is added, it autopopulates the swap fields and the crash no longer happens

**Comment by tharwerth** [comment]
Here's the full rust backtrace for the above error:
```
thread 'main' panicked at src/gui/enter_purge.rs:34:74:
called `Result::unwrap()` on an `Err` value: QueryReturnedNoRows
stack backtrace:
   0:        0x100ff5ee3 - __mh_execute_header
   1:        0x101015e4b - __mh_execute_header
   2:        0x100ff3aee - __mh_execute_header
   3:        0x100ff5cd1 - __mh_execute_header
   4:        0x100ff6f79 - __mh_execute_header
   5:        0x100ff6cf8 - __mh_execute_header
   6:        0x100ff7996 - __mh_execute_header
   7:        0x100ff7302 - __mh_execute_header
   8:        0x100ff63b9 - __mh_execute_header
   9:        0x100ff7036 - __mh_execute_header
  10:        0x101046952 - __mh_execute_header
  11:        0x101046e85 - __mh_execute_header
  12:        0x100b51d5d - __mh_execute_header
  13:        0x100b2ef47 - __mh_execute_header
  14:        0x100b50656 - __mh_execute_header
  15:        0x100b2dc69 - __mh_execute_header
  16:        0x100b5187e - __mh_execute_header
  17:        0x100b4a9e3 - __mh_execute_header
  18:        0x100b2dc69 - __mh_execute_header
  19:        0x100b4b7dd - __mh_execute_header
  20:        0x100b5618f - __mh_execute_header
  21:        0x100c7a4f7 - __mh_execute_header
  22:        0x100c5f858 - __mh_execute_header
  23:        0x100c73843 - __mh_execute_header
  24:        0x100c7c62e - __mh_execute_header
  25:        0x100d328ff - __mh_execute_header
  26:        0x100d30e45 - __mh_execute_header
  27:        0x100d38c6f - __mh_execute_header
  28:     0x7ff808d60ccc - <unknown>
  29:     0x7ff808d60bf1 - <unknown>
  30:     0x7ff808d602a7 - <unknown>
  31:     0x7ff808d5f859 - <unknown>
  32:     0x7ff813c52a09 - <unknown>
  33:     0x7ff813c52816 - <unknown>
  34:     0x7ff813c52561 - <unknown>
  35:     0x7ff80c331c61 - <unknown>
  36:     0x7ff80cc45dc0 - <unknown>
  37:     0x7ff80c323075 - <unknown>
  38:        0x100c9cb32 - __mh_execute_header
  39:        0x100c71cbe - __mh_execute_header
  40:        0x100c99050 - __mh_execute_header
  41:        0x100c7547b - __mh_execute_header
  42:        0x100b5daaf - __mh_execute_header
  43:        0x100b4e2c6 - __mh_execute_header
  44:        0x100b4e2eb - __mh_execute_header
  45:        0x100fedee7 - __mh_execute_header
  46:        0x100b5dedc - __mh_execute_header
thread 'main' panicked at /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.15/src/platform_impl/macos/app_state.rs:387:33:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
stack backtrace:
   0:        0x100ff5ee3 - __mh_execute_header
   1:        0x101015e4b - __mh_execute_header
   2:        0x100ff3aee - __mh_execute_header
   3:        0x100ff5cd1 - __mh_execute_header
   4:        0x100ff6f79 - __mh_execute_header
   5:        0x100ff6cf8 - __mh_execute_header
   6:        0x100ff7996 - __mh_execute_header
   7:        0x100ff7302 - __mh_execute_header
   8:        0x100ff63b9 - __mh_execute_header
   9:        0x100ff7036 - __mh_execute_header
  10:        0x101046952 - __mh_execute_header
  11:        0x101046e85 - __mh_execute_header
  12:        0x100d32bfe - __mh_execute_header
  13:        0x100c71d53 - __mh_execute_header
  14:        0x100c99050 - __mh_execute_header
  15:        0x100c7547b - __mh_execute_header
  16:        0x100b5daaf - __mh_execute_header
  17:        0x100b4e2c6 - __mh_execute_header
  18:        0x100b4e2eb - __mh_execute_header
  19:        0x100fedee7 - __mh_execute_header
  20:        0x100b5dedc - __mh_execute_header
  ```

**Comment by RussHewgill** [comment]
Fixed in [1b3531b](https://github.com/RussHewgill/purgedb/commit/1b3531b5e5aba842f99a61151a6be443e945436e)
